### ISSUE TITLE
Add MLC-LLM integration

### ISSUE THREAD

**Issue by rlouf** [outlines-dev/outlines#735]
OctoML added an [Outlines integration](https://github.com/octoml/mlc-llm/blob/4b59cfa8a677b09d8ee0aaaf4ea4bf98e40635f6/serve/mlc_serve/engine/constrained_sampling.py#L8) in their MLC-LLM fork. Following #728 we could add a MLC-LLM integration that would allow to use `outlines`'s constrained generation via logits processors.
